packages/beacon-node/src/network/reqresp/utils/dataColumnResponseValidaiton.ts

```diff
@@ -0,0 +1,66 @@
+import {LogData} from "@lodestar/logger";
+import {ForkSeq, KZG_COMMITMENTS_GINDEX} from "@lodestar/params";
+import {RespStatus, ResponseError} from "@lodestar/reqresp";
+import {ColumnIndex, Slot} from "@lodestar/types";
+import {Logger, bytesToInt, toHex} from "@lodestar/utils";
+import {IBeaconChain} from "../../../chain/interface.js";
+import {IBeaconDb} from "../../../db/interface.js";
+
+export async function logDataColumnSidecarUnavailability(opts: {
+  chain: IBeaconChain;
+  db: IBeaconDb;
+  index: ColumnIndex;
+  slot: Slot;
+  blockRoot?: Uint8Array;
+  logData: Record<string, LogData>;
+}): Promise<void> {
+  const {chain, db, index, slot, blockRoot, logData} = opts;
+  let localLogData: LogData = {...logData, slot};
+
+  if (blockRoot) {
+    localLogData = {...localLogData, blockRoot: toHex(blockRoot)};
+  }
+
+  const blockBytes = blockRoot ? await db.block.getBinary(blockRoot) : await db.blockArchive.getBinary(slot);
+
+  if (!blockBytes) {
+    chain.logger.error("Expected unfinalized block not found", localLogData);
+    return;
+  }
+
+  const blobsCount =
+    chain.config.getForkSeq(slot) < ForkSeq.deneb
+      ? 0
+      : bytesToInt(blockBytes.slice(KZG_COMMITMENTS_GINDEX, KZG_COMMITMENTS_GINDEX + 4));
```

Comment:
I think we need to check this. From my understanding of ssz, `kzgCommitments` is a variable length field so I think this will give us the byte offset to where the data starts.  We would need to get the offset to the next (variable) container to get the total byte length and then divide by `BYTES_PER_COMMITMENT` to know the array length

Reply:
this should be unit tested